contracts/factory/src/tests/queries.rs

```diff
@@ -390,3 +391,232 @@ fn test_queries_by_tuple_errors() {
+    }
+
+    if token4.address < token3.address {
+        std::mem::swap(&mut token3, &mut token4);
+    }
+
+    if token6.address < token5.address {
+        std::mem::swap(&mut token5, &mut token6);
+    }
+
+    token1.mint(&user_1, &10_000i128);
+    let user1_token1_balance: i128 = env.invoke_contract(
+        &token1.address,
+        &Symbol::new(&env, "balance"),
+        vec![&env, user_1.into_val(&env)],
+    );
+
+    assert_eq!(user1_token1_balance, 10_000i128);
+
+    token2.mint(&user_1, &10_000i128);
+    let user1_token2_balance: i128 = env.invoke_contract(
+        &token2.address,
+        &Symbol::new(&env, "balance"),
+        vec![&env, user_1.into_val(&env)],
+    );
+
+    assert_eq!(user1_token2_balance, 10_000i128);
+
+    let factory = deploy_factory_contract(&env, Some(admin.clone()));
+
+    let first_token_init_info = TokenInitInfo {
+        token_a: token1.address.clone(),
+        token_b: token2.address.clone(),
+    };
+    let first_stake_init_info = StakeInitInfo {
+        min_bond: 1i128,
+        min_reward: 1i128,
+        manager,
+    };
+
+    let first_lp_init_info = LiquidityPoolInitInfo {
+        admin: admin.clone(),
+        fee_recipient: fee_recipient.clone(),
+        max_allowed_slippage_bps: 100,
+        max_allowed_spread_bps: 100,
+        swap_fee_bps: 0,
+        max_referral_bps: 0,
+        token_init_info: first_token_init_info.clone(),
+        stake_init_info: first_stake_init_info,
+    };
+
+    let lp_contract_addr = factory.create_liquidity_pool(
+        &admin.clone(),
+        &first_lp_init_info,
+        &String::from_str(&env, "Pool"),
+        &String::from_str(&env, "PHO/BTC"),
+    );
+
+    let first_lp_client = deploy_lp_contract(&env, lp_contract_addr.clone());
+
+    first_lp_client.provide_liquidity(
+        &user_1.clone(),
+        &Some(100i128),
+        &Some(100i128),
+        &Some(100i128),
+        &Some(100i128),
+        &None::<i64>,
+    );
+
+    let stake_address = factory
+        .query_pool_details(&lp_contract_addr)
+        .pool_response
+        .stake_address;
+
+    let result = factory.query_user_portfolio(&user_1, &true);
+    assert_eq!(
+        result,
+        UserPortfolio {
+            lp_portfolio: vec![
+                &env,
+                LpPortfolio {
+                    assets: (
+                        Asset {
+                            address: token1.address,
+                            amount: 100i128
+                        },
+                        Asset {
+                            address: token2.address,
+                            amount: 100i128
+                        }
+                    )
+                }
+            ],
+            stake_portfolio: vec![
+                &env,
+                StakePortfolio {
+                    stake_token: stake_address,
+                    stakes: vec![&env] // no bonds, empty stake
+                }
```

Comment:
What's the point of this test if you didn't even stake? How do you know it's correct?